docs/website/mkdocs.yml

```diff
@@ -117,6 +117,8 @@ nav:
       - TensorFlow Lite: 'bindings/tensorflow-lite.md'
   - 'Extensions':
       - 'extensions/index.md'
+  - 'Reference':
+      - Optimization Options: 'reference/optimization-options.md'
```

Comment:
Another doc in a similar vein: https://github.com/google/iree/blob/main/docs/developers/best_practices.md, with discussions on https://github.com/google/iree/pull/6541 and https://github.com/google/iree/pull/7110.

As more of this stabilizes I'll want to make another pass through that doc and maybe merge it with this one.

Reply:
Probably not merged: even though detailed with information extraneous to most, this is *user* documentation and needs to stay that way. If you find yourself unlucky enough to be paging through it to try to find your issue, the extra notes can be life saving.